src/lib/timing-src-connector.js

```diff
@@ -0,0 +1,218 @@
+import { subscribe } from './event-subscriber';
+
+/** @typedef {import('./timing-src-connector.types').PlayerControls} PlayerControls */
+/** @typedef {import('./timing-object.types').TimingObject} TimingObject */
+/** @typedef {import('./timing-src-connector.types').TimingSrcConnectorOptions} TimingSrcConnectorOptions */
+/** @typedef {(msg: string) => any} Logger */
+/** @typedef {import('timing-object.types').TConnectionState} TConnectionState */
+
+/**
+ * @type {TimingSrcConnectorOptions}
+ *
+ * For details on these properties and their effects, see the typescript definition referenced above.
+ */
+const defaultOptions = {
+    role: 'viewer',
+    autoPlayMuted: true,
+    allowedDrift: 0,
+    maxAllowedDrift: 0.3,
+    minCheckInterval: 0.1,
+    maxRateAdjustment: 0.1,
+    maxTimeToCatchUp: 1
+};
+
+
+/**
+ * There's a proposed W3C spec for the Timing Object which would introduce a new set of APIs that would simplify time-synchronization tasks for browser applications.
+ *
+ * Proposed spec: https://webtiming.github.io/timingobject/
+ * V3 Spec: https://timingsrc.readthedocs.io/en/latest/
+ * Demuxed talk: https://www.youtube.com/watch?v=cZSjDaGDmX8
+ *
+ * This class makes it easy to connect Vimeo.Player to a provided TimingObject via Vimeo.Player.setTimingSrc(myTimingObject, options) and the synchronization will be handled automatically.
+ *
+ * There are 5 general responsibilities in TimingSrcConnector:
+ *
+ * 1. `updatePlayer()` which sets the player's currentTime, playbackRate and pause/play state based on current state of the TimingObject.
+ * 2. `updateTimingObject()` which sets the TimingObject's position and velocity from the player's state.
+ * 3. `playerUpdater` which listens for change events on the TimingObject and will respond by calling updatePlayer.
+ * 4. `timingObjectUpdater` which listens to the player events of seeked, play and pause and will respond by calling `updateTimingObject()`.
+ * 5. `maintainPlaybackPosition` this is code that constantly monitors the player to make sure it's always in sync with the TimingObject. This is needed because videos will
+ * generally not play with precise time accuracy and there will be some drift which becomes more noticeable over longer periods (as noted in the timing-object spec). More details on this method below.
```

Comment:
Nit: this should stay on the same line as above, or the paragraph should be broken on a new sentence.